app/Repositories/PostRepository.php

```diff
@@ -170,6 +170,54 @@ public function reviews($data)
     }
 
     /**
+     * Updates a post's tags when added or deleted.
+     *
+     * @param array $data
+     *
+     * @return
+     */
+    public function tag($data)
+    {
+        $post = Post::where(['id' => $data['post_id']])->first();
+
+        // Check if the post already has the tag.
+        // If so, soft delete. Otherwise, add the tag to the post.
+        if (in_array($data['tag_name'], $post->tagNames())) {
+            $post->untag($data['tag_name']);
+
+            // Create an event when the tag is deleted.
+            if (!in_array($data['tag_name'], $post->tagNames())) {
+                $post->tag_name = $data['tag_name'];
+                $post->admin_northstar_id = $data['admin_northstar_id'];
+                $post->action = 'delete';
+
+                $event = Event::create([
```

Comment:
actually I don't think we can do this because we won't have access to the specific post (that we need info. from to add to `eventable_id` and `content` in the event unless I'm missing something on how to grab the post from the [documentation](https://github.com/rtconner/laravel-tagging/blob/laravel-5/docs/usage-examples.md)?